avatars/animationHelpers.js

```diff
@@ -37,34 +38,35 @@ import {
   crouchMaxTime,
   // useMaxTime,
   aimMaxTime,
+  AnimationNodeType,
+  AnimationLoopType,
   // avatarInterpolationFrameRate,
   // avatarInterpolationTimeDelay,
   // avatarInterpolationNumFrames,
 } from '../constants.js';
+import game from '../game.js';
 
-const localVector = new Vector3();
-const localVector2 = new Vector3();
-const localVector3 = new Vector3();
-const localVector4 = new Vector3();
+// const localVector = new Vector3();
+// const localVector2 = new Vector3();
 
-const localQuaternion = new Quaternion();
-const localQuaternion2 = new Quaternion();
-const localQuaternion3 = new Quaternion();
-const localQuaternion4 = new Quaternion();
-const localQuaternion5 = new Quaternion();
-const localQuaternion6 = new Quaternion();
+// const localQuaternion = new Quaternion();
+// const localQuaternion2 = new Quaternion();
 
-const identityQuaternion = new Quaternion();
+// const identityQuaternion = new Quaternion();
+
+const isDebugger = false; // Used for debug only codes.Don’t create new data structures on the avatar, to not add any more gc sweep depth in product codes.
 
 let animations;
```

Comment:
Is there any way we can delete all of this local animations initialization?

It seems all of this can be done in WASM now and the JS doesn't even need to parse it?

Reply:
If there is some code that needs to read animations in JS then maybe we can create an API for doing that, but don't parse all animations in JS for all avatars.

Reply:
This would also make this file much smaller and easier to reason about.

Reply:
Got it, I'll try to delete them.

Reply:
>Is there any way we can delete all of this local animations initialization?
It seems all of this can be done in WASM now and the JS doesn't even need to parse it?

Yeah, don't need now. Removed the code for parsing `AnimationClip`s and creating `interpolant`s. https://github.com/webaverse/app/pull/3160/commits/f1649da5ff689c237233d1012b25b9c4f83bce8b

>If there is some code that needs to read animations in JS then maybe we can create an API for doing that, but don't parse all animations in JS for all avatars.

Now using `animationsJson.animations` as `animation` directly, instead of parse into `animationClip`s.
And decorate `animationsJson.animations` directly too.
The only place I found is the need of changing `animation.interpolants` to `animation.tracks`. https://github.com/webaverse/app/pull/3160/commits/f1649da5ff689c237233d1012b25b9c4f83bce8b
Elsewhere work fine, the infos of `animationsJson` is enough ( and mostly the same ).

>This would also make this file much smaller and easier to reason about.

Have not make this file much smaller.
Because of the need to support `.metaversefile` configs, can't hardcode animations' names, so most codes still there.